vm/src/deserializer.cpp

```diff
@@ -0,0 +1,42 @@
+#include "deserializer.h"
+#include <cassert>
+#include <stdexcept>
+
+namespace Borges {
+
+Deserializer::Deserializer(std::size_t n, const double* p)
+    : size_(n), data_(p), index_(0)
+{
+    assert(data_);
+}
+
+void
+Deserializer::deserialize(double* p)
+{
+    if (index_ == size_) {
+        throw std::runtime_error("buffer underflow");
+    }
+    double x = data_[index_];
+    ++index_;
+    *p = x;
+}
+
+void
+deserialize(Deserializer& d, bool* p)
+{
+    *p = deserialize<double>(d);
+}
+
+void
+deserialize(Deserializer& d, double* p)
+{
+    return d.deserialize(p);
+}
+
+void
+deserialize(Deserializer& d, std::size_t* p)
+{
+    *p = deserialize<double>(d);
```

Comment:
shouldn't this be deserialize<std::size_t>(d) ?

Reply:
Similarly as above: std::size_t is serialized as a double, so should be deserialized as a double, then cast to std::size_t.

Reply:
similar comment as above,  an explicit cast is preferred

Reply:
See response above.

Reply:
ok, i yield to your style